js/renovation/viz/core/tooltip.tsx

```diff
@@ -1,55 +1,159 @@
+    MozUserSelect: 'auto',
+    WebkitUserSelect: 'auto',
+  } : {};
   return (
-    <g pointerEvents="none">
-      <PathSvgElement
-        d={getCloudPoints(fullSize, correctedCoordinates, angle,
-          { cornerRadius, arrowWidth }, true)}
-        fill={color}
-        stroke={border.color}
-        strokeWidth={border.width}
-        rotate={angle}
-        rotateX={correctedCoordinates.x}
-        rotateY={correctedCoordinates.y}
-      />
-      <g textAnchor="middle" ref={textRef} transform={`translate(${correctedCoordinates.x}, ${correctedCoordinates.y - size.height / 2 - size.y})`}>
-        <TextSvgElement text={text} />
-      </g>
-    </g>
+    <div
+      style={{
+        position: 'absolute',
+        pointerEvents: 'none',
+        left: cloudSize.x,
+        top: cloudSize.y,
+      }}
+    >
+      <RootSvgElement
+        width={cloudSize.width}
+        height={cloudSize.height}
+        styles={{
+          position: 'absolute',
+          ...styles,
+        }}
+      >
+        <defs>
+          <ShadowFilter
+            id={filterId}
+            x="-50%"
+            y="-50%"
+            width="200%"
+            height="200%"
+            blur={shadow.blur}
+            color={shadow.color}
+            offsetX={shadow.offsetX}
+            offsetY={shadow.offsetY}
+            opacity={shadow.opacity}
+          />
+        </defs>
+        <g
+          filter={getFuncIri(filterId)}
+          ref={cloudRef as any}
+          transform={`translate(${-cloudSize.x}, ${-cloudSize.y})`}
+        >
+          <PathSvgElement
+            pointerEvents={pointerEvents}
+            d={d}
+            fill={customizedOptions.color}
+            stroke={customizedOptions.borderColor}
+            strokeWidth={border.strokeWidth}
+            strokeOpacity={border.strokeOpacity}
+            dashStyle={border.dashStyle}
+            opacity={opacity}
+            rotate={angle}
+            rotateX={correctedCoordinates.x}
+            rotateY={correctedCoordinates.y}
+          />
+          {(customizedOptions.html || TooltipTemplate) ? null
+            : (
+              <g
+                textAnchor="middle"
+                ref={textRef as any}
+                transform={`translate(${correctedCoordinates.x}, ${correctedCoordinates.y - textSize.height / 2 - textSize.y})`}
+              >
+                <TextSvgElement
+                  text={customizedOptions.text}
+                  styles={{
+                    fill: customizedOptions.fontColor,
+                    fontFamily: font.family,
+                    fontSize: font.size,
+                    fontWeight: font.weight,
+                    opacity: font.opacity,
+                    pointerEvents,
+                  }}
+                />
+              </g>
+            )}
+        </g>
+      </RootSvgElement>
+      {!(customizedOptions.html || TooltipTemplate) ? null
+        : (
+          <div
+            ref={htmlRef as any}
```

Comment:
```suggestion
            ref={htmlRef}
```
```as any``` casting is unnecessary since generator v.1.0.37 if RefObject type is used